src/app/space/create/pipelines/services/pipelines.service.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  ErrorHandler,
+  Inject,
+  Injectable
+} from '@angular/core';
+import {
+  Headers,
+  Http,
+  Response
+} from '@angular/http';
+
+import {
+  Observable,
+  Subscription
+} from 'rxjs';
+
+import {
+  Logger,
+  Notification,
+  NotificationType
+} from 'ngx-base';
+import { WIT_API_URL } from 'ngx-fabric8-wit';
+import { AuthenticationService } from 'ngx-login-client';
+
+import { NotificationsService } from 'app/shared/notifications.service';
+
+export interface UserServiceResponse {
+  data: UserServiceData;
+}
+
+export interface UserServiceData {
+  attributes: UserServiceAttributes;
+}
+
+export interface UserServiceAttributes {
+  namespaces: UserServiceNamespace[];
+}
+
+export interface UserServiceNamespace {
+  name: string;
+  type: string;
+  'cluster-console-url': string;
+}
+
+@Injectable()
+export class PipelinesService {
+
+  private readonly headers: Headers = new Headers({ 'Content-Type': 'application/json' });
+  private readonly apiUrl: string;
+
+  private consoleObservable: Observable<string>;
+
+  constructor(
+    private readonly http: Http,
+    private readonly auth: AuthenticationService,
+    private readonly logger: Logger,
+    private readonly errorHandler: ErrorHandler,
+    private readonly notifications: NotificationsService,
+    @Inject(WIT_API_URL) private readonly witUrl: string
+  ) {
+    if (this.auth.getToken() != null) {
+      this.headers.set('Authorization', `Bearer ${this.auth.getToken()}`);
+    }
+    this.apiUrl = witUrl + 'user/services';
+  }
+
+  getOpenshiftConsoleUrl(): Observable<string> {
+    if (!this.consoleObservable) {
+      this.consoleObservable = this.http.get(this.apiUrl, { headers: this.headers })
+        .map((resp: Response) => resp.json() as UserServiceResponse)
+        .catch((err: Response) => this.handleHttpError(err))
+        .map((resp: UserServiceResponse) => resp.data.attributes.namespaces)
+        .map((namespaces: UserServiceNamespace[]) => {
+          for (let namespace of namespaces) {
+            if (namespace.name && namespace.type && namespace.type === 'user' && namespace['cluster-console-url']) {
+              return namespace['cluster-console-url'] + 'project/' + namespace.name + '/browse/pipelines';
+            }
+          }
+
+          return '';
+        });
+    }
+
+    return this.consoleObservable;
+  }
```

Comment:
The initial goal was to cache and I think you're right, I'd need to use my own ReplaySubject for this. However on further consideration, caching automatically for a request that isn't polling doesn't necessarily make sense. If a client does want another http request to go through, this would not be possible with the current service. My original reason was to help prevent cases where multiple users of the service result in multiple http requests for the same data. On the Angular side, with the way the service is currently being imported, would there be the same instance used throughout the app? Or separate instances? Do you know?

Reply:
You've given it a component-level provide, so I believe there will be one service instance for each PipelinesComponent, and the service will share the same lifecycle as the component. If you provide the service at the module level instead then there will be one long-lived service instance that is injected into all consumers.